vendor/signals.py

```diff
@@ -63,4 +65,120 @@ def stripe_delete_customer_signal(sender, instance, **kwargs):
 
     processor.stripe_delete_object(processor.stripe.Customer, instance.meta['stripe_id'])
     logger.success(f"stripe_delete_customer_signal instance: {instance.pk} was successfully deleted on Stripe")
-    
+
+
+@receiver(post_save, sender=Offer)
+def stripe_create_offer_signal(sender, instance, created, **kwargs):
+    site_configured_processor = PaymentProcessorSiteConfig(instance.site)
+
+    if site_configured_processor.get_key_value('payment_processor') != SupportedPaymentProcessor.STRIPE.value:
+        logger.error(
+            f"stripe_create_offer_signal instance: {instance.pk} was not created on stripe for site {instance.site} because site is not configured with Stripe")
+        return None
+
+    if 'stripe' in instance.meta:
+        return None
+
+    processor = StripeProcessor(instance.site)
+
+    query_builder = StripeQueryBuilder()
+
+    pk_clause = query_builder.make_clause_template(
+        field='metadata',
+        key='pk',
+        value=instance.pk,
+        operator=query_builder.EXACT_MATCH,
+        next_operator=query_builder.AND
+    )
+
+    metadata_clause = query_builder.make_clause_template(
+        field='metadata',
+        key='site',
+        value=instance.site.domain,
+        operator=query_builder.EXACT_MATCH
+    )
+
+    query = query_builder.build_search_query(processor.stripe.Product, [pk_clause, metadata_clause])
+    query_result = processor.stripe_query_object(processor.stripe.Product, query)
+
+    if not processor.transaction_succeded:
+        logger.error(f"stripe_create_offer_signal: {processor.transactions_info}")
+        return None
+
+    if query_result:
+        coupons = processor.get_coupons()
+        coupon_matches = processor.match_coupons(coupons, instance)
+        product_id = query_result.data[0].id
+        price_id = processor.get_price_id_with_product(product_id)
+        instance.meta['stripe'] = {
+            'product_id': product_id,
+            'price_id': price_id
+        }
+
+        if coupon_matches:
+            instance.meta['coupon_id'] = coupon_matches[0].id
+
+        instance.save()
+
+        if len(coupon_matches) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one coupon found for offer: {instance.pk} on site: {instance.site.domain}")
+
+        if len(query_result.data) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one offer/product found for pk: {instance.pk} on site: {instance.site.domain}")
```

Comment:
cant you just replace this with the function we already for for creating offers?


`processor.update_offers([offer])`